src/db.c

```diff
@@ -38,6 +39,8 @@ static PyTypeObject AlpmDBType;
 
 static void pyalpm_db_dealloc(AlpmDB *self) {
   Py_TYPE(self)->tp_free((PyObject*)self);
+  if (self->handle)
```

Comment:
Is this really needed?

Reply:
I'd rather have this than not. I'm not super sure if we need it but hey...

Reply:
I am actually not sure if a Py_DECREF of NULL is safe hmmm

Reply:
Let's check, as for the next pacman release alpm_db_update requires a handle object.

```
void Py_INCREF(PyObject *o)¶
Increment the reference count for object o. The object must not be NULL; if you aren’t sure that it isn’t NULL, use Py_XINCREF().
```